test/tests/functional/pbs_soft_walltime.py

```diff
@@ -687,3 +709,336 @@ def test_preempt_order(self):
+
+    def test_soft_walltime_resv(self):
+        """
+        Submit a job with soft walltime inside a reservation
+        """
+
+        now = int(time.time())
+
+        a = {'Resource_List.ncpus': 1, 'reserve_start': now + 5,
+             'reserve_end': now + 10}
+        R = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(R)
+        self.scheduler.log_match(
+            rid + ';Reservation Confirmed', max_attempts=5)
+        r1 = rid.split('.')[0]
+
+        j1 = Job(TEST_USER, attrs={ATTR_queue: r1})
+        jid = self.server.submit(j1)
+        # Set soft walltime to greater than reservation end time
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 300})
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 300}, id=jid)
+
+        # verify that the job gets deleted when reservation ends
+        self.server.expect(
+            JOB, 'queue', op=UNSET, id=jid, offset=10, max_attempts=10)
+
+    def test_restart_server(self):
+        """
+        Test that on server restart soft walltime value is reset
+        """
+
+        hook_body = \
+            """import pbs
+e = pbs.event()
+e.job.Resource_List["soft_walltime"] = 5
+e.accept()
+"""
+        a = {'event': 'queuejob', 'enabled': 'True'}
+        self.server.create_import_hook("que", a, hook_body)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 5}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        time.sleep(6)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:10|10')}, id=jid)
+
+        self.server.restart()
+
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 5}, id=jid)
+
+    def test_resv_job_soft_hard2(self):
+        """
+        Test that a job with soft and hard walltime will not conflict with
+        reservtion if hard walltime is less that reservation start time.
+        """
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        now = int(time.time())
+
+        a = {'Resource_List.ncpus': 4, 'reserve_start': now + 65,
+             'reserve_end': now + 240}
+        R = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(R)
+        self.scheduler.log_match(
+            rid + ';Reservation Confirmed', max_attempts=5)
+
+        a = {'Resource_List.ncpus': 4,
+             'Resource_List.walltime': 60}
+        J = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 60})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_soft_backfill_prime2(self):
+        """
+        Test that soft walltime works with prime queue and
+        non-prime queue
```

Comment:
All you are really testing here is that jobs in a nonprime queue will run in nonprimetime and a job in a prime queue will not.  The fact that these jobs have a soft_walltime is orthogonal to what is being tested.

Reply:
actually for this one I was also thinking twice whether to have it or not but then I added it anyways. but I think you are right and we have this covered so I will remove it.